### DISCUSSION TITLE
Error page redirect always 404?

### DISCUSSION THREAD

**Discussion by leuchtdiode** [pimcore/pimcore#8299] [Q&A / Support]
Hi,

in many of my applications I used the error page to handle legacy URLs being redirect to the new URLs with 301. So basically I did this:

I have an action "errorAction" for the page I'm using as the error page of my website. In this action I fetched the REQUEST_URI from $_SERVER and checked if there is any page which has a certain property with this content. If so, I'm returning a RedirectResponse to the full path of the page with 301 as second argument. Looks like this:

```
public function errorAction()
{
    $requestUri = $_SERVER['REQUEST_URI'];

    $pages = Document::getList([
	'condition' => "`type` = 'page'"
    ]);

    foreach ($pages as $page)
    {
	if (($legacyUrl = $page->getProperty('legacy_url')) && $legacyUrl == $requestUri)
	{
		return new RedirectResponse($page->getFullPath(), 301);
	}
    }
}
```

This always worked well with a 301 redirect. I guess for some time (maybe since migrating to Symfony) the code is being executed just fine, but the browser is showing a 404 instead of my 301. There is a short time flashing a white screen with "Redirecting to /..." and then the redirect happens.

Can someone tell me how I get my 301-method working again?

Thx & BR,
Alex

**Comment by leuchtdiode** [comment]
Hi,

no one has any idea?

A solution would be using "Redirects" which we do for now. But I would be curious where this 404 -> 200 comes from? Is this symfony behaviour?

BR,
Alex

**Reply by passioneight** [reply]
I suggest handling redirects [this way](https://pimcore.com/docs/pimcore/current/Development_Documentation/MVC/Routing_and_URLs/Redirects.html).

Also, you probably shouldn't load all `Page`s for performance reasons.

**Reply by leuchtdiode** [reply]
Thanks. That is the solution we came up with too. I was just curious when and especially why my solution stopped working. I think symfony is "the problem", but I don't know.

You are right about performance problems when loading all pages at once. But I implemented this solution only for websites with max. 50 pages.

BR,
Alex

**Reply by passioneight** [reply]
I see. Well, I guess, you just updated from a Pimcore version built on Zend, right? If so, then chances are that's the cause, but it really depends on how your action was called before (i.e., when the code was working).

**Reply by leuchtdiode** [reply]
Right, I invented the logic when pimcore was still built on Zend. It worked well until I found out a few weeks ago that it doesnt any logner 😁
